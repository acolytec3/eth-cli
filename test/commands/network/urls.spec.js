const NetworkUrlsCommand = require('../../../src/commands/network/urls')

describe('networl:urls', () => {
  let stdoutResult

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString('utf8'))))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'network:urls --help' and throw an error.`, async () => {
    await expect(NetworkUrlsCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'network:urls' and match snapshot.`, async () => {
    await NetworkUrlsCommand.run([])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:urls -display=json' and match snapshot.`, async () => {
    await NetworkUrlsCommand.run(['--display=json'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:urls -display=table' and match snapshot.`, async () => {
    await NetworkUrlsCommand.run(['--display=table'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:urls --bar' and throw an error.`, async () => {
    await expect(NetworkUrlsCommand.run(['--bar'])).rejects.toThrow('Unexpected argument: --bar')
  })
})
