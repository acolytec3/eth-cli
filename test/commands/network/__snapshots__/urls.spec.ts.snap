// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`networl:urls Should run 'network:urls -display=json' and match snapshot. 1`] = `
Array [
  "{
  \\"mainnet\\": \\"https://mainnet.infura.io\\",
  \\"ropsten\\": \\"https://ropsten.infura.io\\",
  \\"rinkeby\\": \\"https://rinkeby.infura.io\\",
  \\"rsk\\": \\"https://public-node.rsk.co\\",
  \\"rsk-testnet\\": \\"https://public-node.testnet.rsk.co\\",
  \\"kovan\\": \\"https://kovan.infura.io\\",
  \\"sokol\\": \\"https://sokol.poa.network\\",
  \\"poa\\": \\"https://core.poa.network\\",
  \\"xdai\\": \\"https://dai.poa.network\\"
}
",
]
`;

exports[`networl:urls Should run 'network:urls -display=table' and match snapshot. 1`] = `
Array [
  "┌───────────────────────────┬───────────────────────────┬───────────────────────────┬────────────────────────────┬────────────────────────────────────┬─────────────────────────┬───────────────────────────┬──────────────────────────┬─────────────────────────┐
│ mainnet                   │ ropsten                   │ rinkeby                   │ rsk                        │ rsk-testnet                        │ kovan                   │ sokol                     │ poa                      │ xdai                    │
├───────────────────────────┼───────────────────────────┼───────────────────────────┼────────────────────────────┼────────────────────────────────────┼─────────────────────────┼───────────────────────────┼──────────────────────────┼─────────────────────────┤
│ https://mainnet.infura.io │ https://ropsten.infura.io │ https://rinkeby.infura.io │ https://public-node.rsk.co │ https://public-node.testnet.rsk.co │ https://kovan.infura.io │ https://sokol.poa.network │ https://core.poa.network │ https://dai.poa.network │
└───────────────────────────┴───────────────────────────┴───────────────────────────┴────────────────────────────┴────────────────────────────────────┴─────────────────────────┴───────────────────────────┴──────────────────────────┴─────────────────────────┘
",
]
`;

exports[`networl:urls Should run 'network:urls' and match snapshot. 1`] = `
Array [
  "{
  \\"mainnet\\": \\"https://mainnet.infura.io\\",
  \\"ropsten\\": \\"https://ropsten.infura.io\\",
  \\"rinkeby\\": \\"https://rinkeby.infura.io\\",
  \\"rsk\\": \\"https://public-node.rsk.co\\",
  \\"rsk-testnet\\": \\"https://public-node.testnet.rsk.co\\",
  \\"kovan\\": \\"https://kovan.infura.io\\",
  \\"sokol\\": \\"https://sokol.poa.network\\",
  \\"poa\\": \\"https://core.poa.network\\",
  \\"xdai\\": \\"https://dai.poa.network\\"
}
",
]
`;
